tasks/upload-sbom-to-atlas/upload-sbom-to-atlas.yaml

```diff
@@ -0,0 +1,261 @@
+        
+        find "$sbomsDir" -type f | while read -r filepath; do
+          # Skip non-JSON files
+          file_relpath=$(realpath "$filepath" --relative-base="$sbomsDir")
+          if ! jq empty "$filepath" 2>/dev/null; then
+            echo "$file_relpath: not a valid JSON"
+            continue
+          fi
+        
+          # Extract the format information using jq
+          sbom_format=$(jq -r 'if .bomFormat then .bomFormat else if .spdxVersion then "SPDX" else empty end end' "$filepath")
+        
+          # If sbom_format is empty, it's not a valid SBOM (CycloneDX or SPDX)
+          if [ -z "$sbom_format" ]; then
+            echo "$file_relpath: not a valid SBOM (CycloneDX or SPDX)"
+            continue  # Skip further processing for non-SBOM files
+          fi
+        
+          # The 'id' of each SBOM is checksum of the original content, before (possibly)
+          # downgrading the CycloneDX version. The conversion always updates some metadata
+          # (timestamp, UUID), changing the checksum. To avoid duplication, use the original
+          # checksum.
+          echo "Calculating SBOM ID for $file_relpath"
+          sbom_id="sha256:$(sha256sum "$filepath" | cut -d ' ' -f 1)"
+        
+          echo "Found $sbom_format SBOM: $file_relpath"
+        
+          # Symlink the discovered SBOMS to ${workdir}/${sbom_id}.json so that subsequent steps
+          # don't have to look for them again.
+          sbom_path="$workdir/$sbom_id.json"
+          ln -s "$(realpath "$filepath")" "$sbom_path"
+        
+          # Check if SBOM is CycloneDX or SPDX
+          if [ "$sbom_format" == "CycloneDX" ]; then
+            echo "Found CycloneDX SBOM: $file_relpath"
+        
+            sbom_version="$(jq -r ".specVersion" "$sbom_path")"
+            supportedCycloneDxVersion=$(params.supportedCycloneDxVersion)
+            version_lesser_equal "$sbom_version" "$supportedCycloneDxVersion"
+        
+            printf "%s" "cyclonedx-json" > "${sbom_path}.sbom_format"
+        
+          elif [ "$sbom_format" == "SPDX" ]; then
+            echo "Found SPDX SBOM: $file_relpath"
+        
+            sbom_version="$(jq -r '.spdxVersion | sub("SPDX-"; "")' "$sbom_path")"
+            supportedSpdxVersion=$(params.supportedSpdxVersion)
+            version_lesser_equal "$sbom_version" "$supportedSpdxVersion"
+        
+            printf "%s" "spdx-json" > "${sbom_path}.sbom_format"
+        
+          else
+            echo "$file_relpath: not a valid SBOM (CycloneDX or SPDX)"
+            continue
+          fi
+        done
+        echo "Found $(find "$workdir" -name "*.json" | wc -l) SBOMs"
+
+    # Needs syft, which is in a different image => has to be a separate step
+    - name: convert-sboms-if-needed
+      image: registry.redhat.io/rh-syft-tech-preview/syft-rhel9:1.0.1@sha256:27c268d678103a27b6964c2cd5169040941b7304d0078f9727789ffb8ffba370
```

Comment:
This is problematic because this needs a token to pull and so github ci won't be able to run the tests. Maybe we should add syft to the utils image?

Reply:
Sure no problem, I just created PR [here](https://github.com/konflux-ci/release-service-utils/pull/264).